test/buffer.jl

```diff
@@ -0,0 +1,70 @@
+using Serialization: deserialize, serialize
+using Test
+using ray_core_worker_julia_jll: LocalMemoryBuffer, Data, Size, OwnsData, IsPlasmaBuffer
+
+@testset "LocalMemoryBuffer" begin
+    data = UInt16[1:3;]
+
+    @testset "non-copied object reference" begin
+        buffer = LocalMemoryBuffer(pointer_from_objref(data), sizeof(data), false)
+        @test Data(buffer[]) == pointer_from_objref(data)
+        @test Size(buffer[]) == sizeof(data)
+        @test !OwnsData(buffer[])
+        @test !IsPlasmaBuffer(buffer[])
+
+        buffer_ptr = Data(buffer[]).cpp_object
```

Comment:
> If we added a `unsafe_pointer_to_objref(::CxxPtr)`

I think this would be useful to do, maybe as an upstream PR to CxxWrap?  referring to the wrapped pointer seems dangerous.